### PR TITLE
Post sales

### PR DESCRIPTION
**Pull request by zhirzh**
storybook: https://post-sales--5cb6f92366a0080020cba496.chromatic.com

## Input

- add css overrides: loadingClassName, highlightClassName
- only render label if placeholder is present
- export types: CommonInputProps, InputType

## Select

- add css overrides: arrowClassName
- export types: CommonSelectProps

## Sidebar

- hide scrollbars if content not scrollable

## Modal

- hide scrollbars if content not scrollable

### CODE REVIEW

File: packages/pebble-web/src/components/Select.tsx [hunk 1]
```diff
@@ -48,9 +49,15 @@ function Select<OptionType>(props: SelectProps<OptionType>) {
...
         labelComponent={({ toggleDropdown, isOpen }) => {
-          const chevron = cx(chevronStyle, "pi", "pi-arrow-drop-down", {
-            __pebble__select__open: isOpen
-          });
+          const chevron = cx(
+            chevronStyle,
+            "pi",
+            "pi-arrow-drop-down",
+            {
+              __pebble__select__open: isOpen
```
Comment (venky-0710) [inline packages/pebble-web/src/components/Select.tsx:57]: Where are you getting these class names , i have seen this in the chevron style as well

File: packages/pebble-web/src/components/Input.tsx [hunk 2]
```diff
@@ -103,12 +104,16 @@ class Input extends React.PureComponent<InputProps, InputState> {
...
-    const highlightClassName = cx(highlightStyle, {
-      _pebble_input_highlight_focused: !!isFocused,
-      _pebble_input_highlight_state: !!errorMessage || !!successMessage,
-      _pebble_input_highlight_read_only: !!readOnly,
-      _pebble_input_highlight_disabled: !!disabled
-    });
+    const _highlightClassName = cx(
+      highlightStyle,
+      {
+        _pebble_input_highlight_focused: !!isFocused,
```
Comment (venky-0710) [inline packages/pebble-web/src/components/Input.tsx:110]: How are you getting these class names

File: packages/pebble-web/src/components/__tests__/__snapshots__/phonenumberinput.test.tsx.snap [hunk 3]
```diff
@@ -1,7 +1,7 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
 
 exports[`Component: Select default snapshot 1`] = `
-.emotion-12 {
+.emotion-10 {
```
Comment (venky-0710) [inline packages/pebble-web/src/components/__tests__/__snapshots__/phonenumberinput.test.tsx.snap:4]: I did not understand what is that you are trying to do , like are you trying to downgrade the emotion version or something.

**Review by venky-0710 — COMMENTED** [review]
I did not understand how you are getting __pebble classnames ,

File: packages/pebble-web/src/components/Input.tsx [hunk 4]
```diff
@@ -100,15 +102,19 @@ class Input extends React.PureComponent<InputProps, InputState> {
       disabled,
       onChange: this.handleChange,
       readOnly,
-      value: value || ""
+      value: value === 0 ? "0" : value || ""
```
Comment (vkajith) [inline packages/pebble-web/src/components/Input.tsx:105]:
```suggestion
      value: value || ""
```

File: packages/pebble-web/src/components/__tests__/__snapshots__/phonenumberinput.test.tsx.snap (hunk 3 above)
Comment (zhirzh) [reply packages/pebble-web/src/components/__tests__/__snapshots__/phonenumberinput.test.tsx.snap:4]: `.snap` files are generated by [jest snapshot tests](https://jestjs.io/docs/snapshot-testing)

**Review by zhirzh — COMMENTED** [review]
@venky-0710 revert this commit: bf0e143

#427 is not related to post sales. it is a general bugfix in pebble

File: packages/pebble-web/src/components/typings/DateInput.ts [hunk 5]
```diff
@@ -11,6 +11,7 @@ export interface DateInputProps {
   disabled?: boolean;
   placement?: Placement;
   wrapperClassName?: string;
+  dropDownWrapperClassName?:string
```
Comment (zhirzh) [inline packages/pebble-web/src/components/typings/DateInput.ts:14]: when overriding internal styles, prefer using target className as prop

```suggestion
  dropDownClassName?:string
```

**Review by appy9 — APPROVED** [review]

**Review by venky-0710 — APPROVED** [review]

**Pull request merged**